conkit-plot peval fails to match sequences

## General Information

<!-- Can be obtained by running: ``python -c "import conkit; print(conkit.__version__)"`` -->
- ConKit version: 0.12.0
- Python version: 3.6.9
- Environment (if applicable): WSL
<!-- Such as CCP4, Anaconda, CCP-EM, etc. -->

## Example

A minimal example to reproduce the error:
```
../.local/bin/conkit-plot cmap conkit.jones jones conkit.rr casprr
or
../.local/bin/conkit-plot cmap conkit.jones jones conkit.mat ccmpred
```
The .jones, .rr and .mat file were generated from the conkit-predict script, which worked fine. I used them to exclude the possibility that my original files were the problem; my original goal was use a reference structure to determine which contacts were true positives, but I was getting a traceback. So I tried just plotting without the reference structure and I got the same traceback.

## Traceback

The Python traceback
```
File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/sadiogo/.local/lib/python3.6/site-packages/conkit/command_line/conkit_plot.py", line 441, in <module>
    main()
  File "/home/sadiogo/.local/lib/python3.6/site-packages/conkit/command_line/conkit_plot.py", line 283, in main
    con.set_sequence_register()
  File "/home/sadiogo/.local/lib/python3.6/site-packages/conkit/core/contactmap.py", line 503, in set_sequence_register
    c.res1 = self.sequence.seq[res1_index - 1]
  File "/home/sadiogo/.local/lib/python3.6/site-packages/conkit/core/contact.py", line 268, in res1
    self._res1 = Contact._set_residue(amino_acid)
  File "/home/sadiogo/.local/lib/python3.6/site-packages/conkit/core/contact.py", line 483, in _set_residue
    raise ValueError("Unknown amino acid: {} (assert all is uppercase!)".format(amino_acid))
ValueError: Unknown amino acid: - (assert all is uppercase!)
```

I've installed python 3.8 and attempted to install conkit, but I get the following error:
```
Building wheels for collected packages: conkit
  Building wheel for conkit (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /usr/bin/python3 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-ifik9dmj/conkit_44e39083b73346fcb021bb7f44fbdbeb/setup.py'"'"'; __file__='"'"'/tmp/pip-install-ifik9dmj/conkit_44e39083b73346fcb021bb7f44fbdbeb/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-ykdbqezn
       cwd: /tmp/pip-install-ifik9dmj/conkit_44e39083b73346fcb021bb7f44fbdbeb/
  Complete output (116 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.8
  creating build/lib.linux-x86_64-3.8/conkit
  copying conkit/version.py -> build/lib.linux-x86_64-3.8/conkit
  copying conkit/__init__.py -> build/lib.linux-x86_64-3.8/conkit
  creating build/lib.linux-x86_64-3.8/conkit/applications
  copying conkit/applications/cdhit.py -> build/lib.linux-x86_64-3.8/conkit/applications
  copying conkit/applications/ccmpred.py -> build/lib.linux-x86_64-3.8/conkit/applications
  copying conkit/applications/bbcontacts.py -> build/lib.linux-x86_64-3.8/conkit/applications
  copying conkit/applications/hhblits.py -> build/lib.linux-x86_64-3.8/conkit/applications
  copying conkit/applications/hhfilter.py -> build/lib.linux-x86_64-3.8/conkit/applications
  copying conkit/applications/psicov.py -> build/lib.linux-x86_64-3.8/conkit/applications
  copying conkit/applications/__init__.py -> build/lib.linux-x86_64-3.8/conkit/applications
  copying conkit/applications/jackhmmer.py -> build/lib.linux-x86_64-3.8/conkit/applications
  creating build/lib.linux-x86_64-3.8/conkit/command_line
  copying conkit/command_line/conkit_msatool.py -> build/lib.linux-x86_64-3.8/conkit/command_line
  copying conkit/command_line/conkit_convert.py -> build/lib.linux-x86_64-3.8/conkit/command_line
  copying conkit/command_line/conkit_precision.py -> build/lib.linux-x86_64-3.8/conkit/command_line
  copying conkit/command_line/conkit_predict.py -> build/lib.linux-x86_64-3.8/conkit/command_line
  copying conkit/command_line/conkit_plot.py -> build/lib.linux-x86_64-3.8/conkit/command_line
  copying conkit/command_line/__init__.py -> build/lib.linux-x86_64-3.8/conkit/command_line
  creating build/lib.linux-x86_64-3.8/conkit/core
  copying conkit/core/entity.py -> build/lib.linux-x86_64-3.8/conkit/core
  copying conkit/core/sequence.py -> build/lib.linux-x86_64-3.8/conkit/core
  copying conkit/core/sequencefile.py -> build/lib.linux-x86_64-3.8/conkit/core
  copying conkit/core/contactmap.py -> build/lib.linux-x86_64-3.8/conkit/core
  copying conkit/core/mappings.py -> build/lib.linux-x86_64-3.8/conkit/core
  copying conkit/core/struct.py -> build/lib.linux-x86_64-3.8/conkit/core
  copying conkit/core/__init__.py -> build/lib.linux-x86_64-3.8/conkit/core
  copying conkit/core/contact.py -> build/lib.linux-x86_64-3.8/conkit/core
  copying conkit/core/contactfile.py -> build/lib.linux-x86_64-3.8/conkit/core
  creating build/lib.linux-x86_64-3.8/conkit/core/ext
  copying conkit/core/ext/__init__.py -> build/lib.linux-x86_64-3.8/conkit/core/ext
  creating build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/membrain.py -> build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/comsat.py -> build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/ncont.py -> build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/pcons.py -> build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/casp.py -> build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/_parser.py -> build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/ccmpred.py -> build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/jones.py -> build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/epcmap.py -> build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/stockholm.py -> build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/mapalign.py -> build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/fasta.py -> build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/pdb.py -> build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/rosetta.py -> build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/clustal.py -> build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/freecontact.py -> build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/aleigen.py -> build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/bbcontacts.py -> build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/a2m.py -> build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/bclcontact.py -> build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/a3m.py -> build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/plmdca.py -> build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/psicov.py -> build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/evfold.py -> build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/__init__.py -> build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/gremlin.py -> build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/_iotools.py -> build/lib.linux-x86_64-3.8/conkit/io
  copying conkit/io/_cache.py -> build/lib.linux-x86_64-3.8/conkit/io
  creating build/lib.linux-x86_64-3.8/conkit/misc
  copying conkit/misc/energyfunction.py -> build/lib.linux-x86_64-3.8/conkit/misc
  copying conkit/misc/distances.py -> build/lib.linux-x86_64-3.8/conkit/misc
  copying conkit/misc/bandwidth.py -> build/lib.linux-x86_64-3.8/conkit/misc
  copying conkit/misc/selector.py -> build/lib.linux-x86_64-3.8/conkit/misc
  copying conkit/misc/__init__.py -> build/lib.linux-x86_64-3.8/conkit/misc
  copying conkit/misc/selectalg.py -> build/lib.linux-x86_64-3.8/conkit/misc
  creating build/lib.linux-x86_64-3.8/conkit/misc/ext
  copying conkit/misc/ext/__init__.py -> build/lib.linux-x86_64-3.8/conkit/misc/ext
  creating build/lib.linux-x86_64-3.8/conkit/plot
  copying conkit/plot/precisionevaluation.py -> build/lib.linux-x86_64-3.8/conkit/plot
  copying conkit/plot/contactmapchord.py -> build/lib.linux-x86_64-3.8/conkit/plot
  copying conkit/plot/contactmapmatrix.py -> build/lib.linux-x86_64-3.8/conkit/plot
  copying conkit/plot/figure.py -> build/lib.linux-x86_64-3.8/conkit/plot
  copying conkit/plot/contactdensity.py -> build/lib.linux-x86_64-3.8/conkit/plot
  copying conkit/plot/tools.py -> build/lib.linux-x86_64-3.8/conkit/plot
  copying conkit/plot/contactmap.py -> build/lib.linux-x86_64-3.8/conkit/plot
  copying conkit/plot/sequencecoverage.py -> build/lib.linux-x86_64-3.8/conkit/plot
  copying conkit/plot/__init__.py -> build/lib.linux-x86_64-3.8/conkit/plot
  running egg_info
  writing conkit.egg-info/PKG-INFO
  writing dependency_links to conkit.egg-info/dependency_links.txt
  writing requirements to conkit.egg-info/requires.txt
  writing top-level names to conkit.egg-info/top_level.txt
  reading manifest file 'conkit.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  writing manifest file 'conkit.egg-info/SOURCES.txt'
  copying conkit/core/ext/c_contactmap.pyx -> build/lib.linux-x86_64-3.8/conkit/core/ext
  copying conkit/core/ext/c_sequencefile.pyx -> build/lib.linux-x86_64-3.8/conkit/core/ext
  copying conkit/misc/ext/c_bandwidth.pyx -> build/lib.linux-x86_64-3.8/conkit/misc/ext
  running build_ext
  cythoning conkit/core/ext/c_contactmap.pyx to conkit/core/ext/c_contactmap.c
  /home/sadiogo/.local/lib/python3.8/site-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /tmp/pip-install-ifik9dmj/conkit_44e39083b73346fcb021bb7f44fbdbeb/conkit/core/ext/c_contactmap.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)
  cythoning conkit/core/ext/c_sequencefile.pyx to conkit/core/ext/c_sequencefile.c
  /home/sadiogo/.local/lib/python3.8/site-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /tmp/pip-install-ifik9dmj/conkit_44e39083b73346fcb021bb7f44fbdbeb/conkit/core/ext/c_sequencefile.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)
  cythoning conkit/misc/ext/c_bandwidth.pyx to conkit/misc/ext/c_bandwidth.c
  /home/sadiogo/.local/lib/python3.8/site-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /tmp/pip-install-ifik9dmj/conkit_44e39083b73346fcb021bb7f44fbdbeb/conkit/misc/ext/c_bandwidth.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)
  building 'conkit.core.ext.c_contactmap' extension
  creating build/temp.linux-x86_64-3.8
  creating build/temp.linux-x86_64-3.8/conkit
  creating build/temp.linux-x86_64-3.8/conkit/core
  creating build/temp.linux-x86_64-3.8/conkit/core/ext
  x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/home/sadiogo/.local/lib/python3.8/site-packages/numpy/core/include -I/usr/include/python3.8 -c conkit/core/ext/c_contactmap.c -o build/temp.linux-x86_64-3.8/conkit/core/ext/c_contactmap.o -O3 -ffast-math -march=native -pipe -fopenmp
  conkit/core/ext/c_contactmap.c:6:10: fatal error: Python.h: No such file or directory
   #include "Python.h"
            ^~~~~~~~~~
  compilation terminated.
  error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for conkit
Running setup.py clean for conkit
Failed to build conkit
Installing collected packages: conkit
    Running setup.py install for conkit ... error
    ERROR: Command errored out with exit status 1:
     command: /usr/bin/python3 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-ifik9dmj/conkit_44e39083b73346fcb021bb7f44fbdbeb/setup.py'"'"'; __file__='"'"'/tmp/pip-install-ifik9dmj/conkit_44e39083b73346fcb021bb7f44fbdbeb/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-y3xtj15b/install-record.txt --single-version-externally-managed --user --prefix= --compile --install-headers /home/sadiogo/.local/include/python3.8/conkit
         cwd: /tmp/pip-install-ifik9dmj/conkit_44e39083b73346fcb021bb7f44fbdbeb/
    Complete output (110 lines):
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.8
    creating build/lib.linux-x86_64-3.8/conkit
    copying conkit/version.py -> build/lib.linux-x86_64-3.8/conkit
    copying conkit/__init__.py -> build/lib.linux-x86_64-3.8/conkit
    creating build/lib.linux-x86_64-3.8/conkit/applications
    copying conkit/applications/cdhit.py -> build/lib.linux-x86_64-3.8/conkit/applications
    copying conkit/applications/ccmpred.py -> build/lib.linux-x86_64-3.8/conkit/applications
    copying conkit/applications/bbcontacts.py -> build/lib.linux-x86_64-3.8/conkit/applications
    copying conkit/applications/hhblits.py -> build/lib.linux-x86_64-3.8/conkit/applications
    copying conkit/applications/hhfilter.py -> build/lib.linux-x86_64-3.8/conkit/applications
    copying conkit/applications/psicov.py -> build/lib.linux-x86_64-3.8/conkit/applications
    copying conkit/applications/__init__.py -> build/lib.linux-x86_64-3.8/conkit/applications
    copying conkit/applications/jackhmmer.py -> build/lib.linux-x86_64-3.8/conkit/applications
    creating build/lib.linux-x86_64-3.8/conkit/command_line
    copying conkit/command_line/conkit_msatool.py -> build/lib.linux-x86_64-3.8/conkit/command_line
    copying conkit/command_line/conkit_convert.py -> build/lib.linux-x86_64-3.8/conkit/command_line
    copying conkit/command_line/conkit_precision.py -> build/lib.linux-x86_64-3.8/conkit/command_line
    copying conkit/command_line/conkit_predict.py -> build/lib.linux-x86_64-3.8/conkit/command_line
    copying conkit/command_line/conkit_plot.py -> build/lib.linux-x86_64-3.8/conkit/command_line
    copying conkit/command_line/__init__.py -> build/lib.linux-x86_64-3.8/conkit/command_line
    creating build/lib.linux-x86_64-3.8/conkit/core
    copying conkit/core/entity.py -> build/lib.linux-x86_64-3.8/conkit/core
    copying conkit/core/sequence.py -> build/lib.linux-x86_64-3.8/conkit/core
    copying conkit/core/sequencefile.py -> build/lib.linux-x86_64-3.8/conkit/core
    copying conkit/core/contactmap.py -> build/lib.linux-x86_64-3.8/conkit/core
    copying conkit/core/mappings.py -> build/lib.linux-x86_64-3.8/conkit/core
    copying conkit/core/struct.py -> build/lib.linux-x86_64-3.8/conkit/core
    copying conkit/core/__init__.py -> build/lib.linux-x86_64-3.8/conkit/core
    copying conkit/core/contact.py -> build/lib.linux-x86_64-3.8/conkit/core
    copying conkit/core/contactfile.py -> build/lib.linux-x86_64-3.8/conkit/core
    creating build/lib.linux-x86_64-3.8/conkit/core/ext
    copying conkit/core/ext/__init__.py -> build/lib.linux-x86_64-3.8/conkit/core/ext
    creating build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/membrain.py -> build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/comsat.py -> build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/ncont.py -> build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/pcons.py -> build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/casp.py -> build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/_parser.py -> build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/ccmpred.py -> build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/jones.py -> build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/epcmap.py -> build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/stockholm.py -> build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/mapalign.py -> build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/fasta.py -> build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/pdb.py -> build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/rosetta.py -> build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/clustal.py -> build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/freecontact.py -> build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/aleigen.py -> build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/bbcontacts.py -> build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/a2m.py -> build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/bclcontact.py -> build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/a3m.py -> build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/plmdca.py -> build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/psicov.py -> build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/evfold.py -> build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/__init__.py -> build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/gremlin.py -> build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/_iotools.py -> build/lib.linux-x86_64-3.8/conkit/io
    copying conkit/io/_cache.py -> build/lib.linux-x86_64-3.8/conkit/io
    creating build/lib.linux-x86_64-3.8/conkit/misc
    copying conkit/misc/energyfunction.py -> build/lib.linux-x86_64-3.8/conkit/misc
    copying conkit/misc/distances.py -> build/lib.linux-x86_64-3.8/conkit/misc
    copying conkit/misc/bandwidth.py -> build/lib.linux-x86_64-3.8/conkit/misc
    copying conkit/misc/selector.py -> build/lib.linux-x86_64-3.8/conkit/misc
    copying conkit/misc/__init__.py -> build/lib.linux-x86_64-3.8/conkit/misc
    copying conkit/misc/selectalg.py -> build/lib.linux-x86_64-3.8/conkit/misc
    creating build/lib.linux-x86_64-3.8/conkit/misc/ext
    copying conkit/misc/ext/__init__.py -> build/lib.linux-x86_64-3.8/conkit/misc/ext
    creating build/lib.linux-x86_64-3.8/conkit/plot
    copying conkit/plot/precisionevaluation.py -> build/lib.linux-x86_64-3.8/conkit/plot
    copying conkit/plot/contactmapchord.py -> build/lib.linux-x86_64-3.8/conkit/plot
    copying conkit/plot/contactmapmatrix.py -> build/lib.linux-x86_64-3.8/conkit/plot
    copying conkit/plot/figure.py -> build/lib.linux-x86_64-3.8/conkit/plot
    copying conkit/plot/contactdensity.py -> build/lib.linux-x86_64-3.8/conkit/plot
    copying conkit/plot/tools.py -> build/lib.linux-x86_64-3.8/conkit/plot
    copying conkit/plot/contactmap.py -> build/lib.linux-x86_64-3.8/conkit/plot
    copying conkit/plot/sequencecoverage.py -> build/lib.linux-x86_64-3.8/conkit/plot
    copying conkit/plot/__init__.py -> build/lib.linux-x86_64-3.8/conkit/plot
    running egg_info
    writing conkit.egg-info/PKG-INFO
    writing dependency_links to conkit.egg-info/dependency_links.txt
    writing requirements to conkit.egg-info/requires.txt
    writing top-level names to conkit.egg-info/top_level.txt
    reading manifest file 'conkit.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    writing manifest file 'conkit.egg-info/SOURCES.txt'
    copying conkit/core/ext/c_contactmap.pyx -> build/lib.linux-x86_64-3.8/conkit/core/ext
    copying conkit/core/ext/c_sequencefile.pyx -> build/lib.linux-x86_64-3.8/conkit/core/ext
    copying conkit/misc/ext/c_bandwidth.pyx -> build/lib.linux-x86_64-3.8/conkit/misc/ext
    running build_ext
    skipping 'conkit/core/ext/c_contactmap.c' Cython extension (up-to-date)
    skipping 'conkit/core/ext/c_sequencefile.c' Cython extension (up-to-date)
    skipping 'conkit/misc/ext/c_bandwidth.c' Cython extension (up-to-date)
    building 'conkit.core.ext.c_contactmap' extension
    creating build/temp.linux-x86_64-3.8
    creating build/temp.linux-x86_64-3.8/conkit
    creating build/temp.linux-x86_64-3.8/conkit/core
    creating build/temp.linux-x86_64-3.8/conkit/core/ext
    x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/home/sadiogo/.local/lib/python3.8/site-packages/numpy/core/include -I/usr/include/python3.8 -c conkit/core/ext/c_contactmap.c -o build/temp.linux-x86_64-3.8/conkit/core/ext/c_contactmap.o -O3 -ffast-math -march=native -pipe -fopenmp
    conkit/core/ext/c_contactmap.c:6:10: fatal error: Python.h: No such file or directory
     #include "Python.h"
              ^~~~~~~~~~
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /usr/bin/python3 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-ifik9dmj/conkit_44e39083b73346fcb021bb7f44fbdbeb/setup.py'"'"'; __file__='"'"'/tmp/pip-install-ifik9dmj/conkit_44e39083b73346fcb021bb7f44fbdbeb/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-y3xtj15b/install-record.txt --single-version-externally-managed --user --prefix= --compile --install-headers /home/sadiogo/.local/include/python3.8/conkit Check the logs for full command output.
```

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -15,7 +15,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        python-version: [3.6, 3.7, 3.8, 3.9]
+        python-version: [3.7, 3.8, 3.9]
 
     steps:
     - uses: actions/checkout@v1

--- a/conkit/command_line/conkit_plot.py
+++ b/conkit/command_line/conkit_plot.py
@@ -516,10 +516,12 @@ def main(argv=None):
         con.sort("raw_score", reverse=True, inplace=True)
 
         if args.pdbchain:
-            pdb = conkit.io.read(args.pdbfile, "pdb")[args.pdbchain]
+            pdb = conkit.io.read(args.pdbfile, args.pdbformat)[args.pdbchain]
         else:
-            pdb = conkit.io.read(args.pdbfile, "pdb")[0]
+            pdb = conkit.io.read(args.pdbfile, args.pdbformat)[0]
 
+        pdb.sequence = seq
+        pdb.set_sequence_register()
         pdb = pdb.as_contactmap()
         con_matched = con.match(pdb, renumber=True, remove_unmatched=True)
 

--- a/conkit/command_line/conkit_precision.py
+++ b/conkit/command_line/conkit_precision.py
@@ -79,9 +79,14 @@ def main():
         pdb = conkit.io.read(args.pdbfile, args.pdbformat)[args.pdbchain]
     else:
         pdb = conkit.io.read(args.pdbfile, args.pdbformat)[0]
-    seq = conkit.io.read(args.seqfile, args.seqformat)[0]
-    con = conkit.io.read(args.confile, args.conformat)[0]
 
+    seq = conkit.io.read(args.seqfile, args.seqformat)[0]
+
+    pdb.sequence = seq
+    pdb.set_sequence_register()
+    pdb = pdb.as_contactmap()
+
+    con = conkit.io.read(args.confile, args.conformat)[0]
     con.sequence = seq
     con.set_sequence_register()
 

--- a/conkit/command_line/conkit_predict.py
+++ b/conkit/command_line/conkit_predict.py
@@ -177,7 +177,7 @@ def main(argl=None):
         conkit.io.convert(a3m_fname, "a3m", jon_fname, "jones")
 
     else:
-        raise RuntimeError("Should never get to here")
+        raise RuntimeError("Error reading cli arguments - please report this bug")
 
     # CCMpred requires alignments to be in the *jones* format - i.e. the format created
     # and used by David Jones in PSICOV

--- a/conkit/core/contactmap.py
+++ b/conkit/core/contactmap.py
@@ -831,7 +831,7 @@ class ContactMap(Entity):
                 else:
                     contact_map1[_id].status = ContactMatchState.false_positive
             else:
-                raise RuntimeError("Error matching two contact maps - this should never happen")
+                raise RuntimeError("Error matching two contact maps - please report this bug")
 
         # ================================================================
         # 3. Add false negatives
@@ -1164,6 +1164,6 @@ class ContactMap(Entity):
                     contact.res2_seq = other_residue.res_seq
                     contact.res2_chain = other_residue.res_chain
                 else:
-                    raise ValueError("Should never get here")
+                    raise ValueError("Error renumbering contact map - please report this bug")
 
         return contact_map

--- a/conkit/core/contactmap.py
+++ b/conkit/core/contactmap.py
@@ -539,7 +539,7 @@ class ContactMap(Entity):
             else:
                 res1_index = c.res1_seq
                 res2_index = c.res2_seq
-            if res1_index <= seq_len or res2_index <= seq_len:
+            if res1_index <= seq_len and res2_index <= seq_len:
                 c.res1 = self.sequence.seq[res1_index - 1]
                 c.res2 = self.sequence.seq[res2_index - 1]
             else:

--- a/conkit/core/contactmap.py
+++ b/conkit/core/contactmap.py
@@ -530,6 +530,8 @@ class ContactMap(Entity):
         if self.sequence is None:
             raise ValueError("No sequence defined")
 
+        seq_len = len(self.sequence)
+
         for c in self:
             if altloc:
                 res1_index = c.res1_altseq
@@ -537,8 +539,11 @@ class ContactMap(Entity):
             else:
                 res1_index = c.res1_seq
                 res2_index = c.res2_seq
-            c.res1 = self.sequence.seq[res1_index - 1]
-            c.res2 = self.sequence.seq[res2_index - 1]
+            if res1_index <= seq_len or res2_index <= seq_len:
+                c.res1 = self.sequence.seq[res1_index - 1]
+                c.res2 = self.sequence.seq[res2_index - 1]
+            else:
+                raise ValueError('Contact {} is out of sequence bounds'.format(c.id))
 
     @deprecate("0.11", msg="Use get_jaccard_index instead.")
     def calculate_jaccard_index(self, other):

--- a/conkit/core/tests/test_contactmap.py
+++ b/conkit/core/tests/test_contactmap.py
@@ -432,6 +432,22 @@ class TestContactMap(unittest.TestCase):
             [("A", "E"), ("B", "F"), ("A", "D"), ("C", "F"), ("B", "E")], [(c.res1, c.res2) for c in contact_map1]
         )
 
+    def test_set_sequence_register_3(self):
+        contact_map1 = ContactMap("1")
+        for comb in [(1, 5, 1.0), (2, 6, 1.0), (1, 4, 1.0), (3, 6, 1.0), (2, 5, 1.0)]:
+            contact_map1.add(Contact(*comb))
+        contact_map1.sequence = Sequence("foo", "ABCDE")
+        with self.assertRaises(ValueError):
+            contact_map1.set_sequence_register()
+
+    def test_set_sequence_register_4(self):
+        contact_map1 = ContactMap("1")
+        for comb in [(1, 5, 1.0), (6, 3, 1.0), (1, 4, 1.0), (3, 2, 1.0), (2, 5, 1.0)]:
+            contact_map1.add(Contact(*comb))
+        contact_map1.sequence = Sequence("foo", "ABCDE")
+        with self.assertRaises(ValueError):
+            contact_map1.set_sequence_register()
+
     def test_match_1(self):
         contact_map1 = ContactMap("foo")
         for params in [(1, 5, 1.0), (1, 6, 1.0), (2, 7, 1.0), (3, 5, 1.0), (2, 8, 1.0)]:

